### ISSUE TITLE
Change session-based script interface to use separate subcommands.

### ISSUE THREAD

**Issue by riccardomurri** [uzh/gc3pie#533]
At the moment, a session-based script provides one and the same
command-line interface to create a session, manage, and to run it.

This was intentional: the code for the first session-based scripts
would recompute the list of jobs upon invocation, add new jobs to the
session, and then start the "run" phase.  The rationale was that
people would get consistent results by always entering the same
command-line.

However, at some point in the history we changed this[^1]: if a
session already exists, the script proceeds to the running step and
may not update the list of jobs based on changed command-line
parameters; this warning message is printed:

        WARNING: Session already exists, some command-line arguments might be ignored.

Based on the new behavior, the "unified command-line" interface does
not make much sense anymore.  Likewise, options like `-N` now look awkward.

I would thus suggest that we change `SessionBasedScript` to use sub-commands:

        # create a session; params are needed to specify what and how
        script new SESSION_NAME ...

        # run a session -- no params needed, everything is picked up from session
        script run SESSION_NAME

We might also consider merging `gsession` into session-based script...

[^1]: This is the commit message that introduced the change:

        commit 7a9de260cc167a6c4472eda8b7894301ffb100eb
        Author: riccardo.murri <riccardo.murri@d0f2d5a0-84c0-18d2-3862-8add1af26540>
        Date:   Sat Sep 29 18:10:27 2012 +0000

            Allow `new_tasks()` in `SessionBasedScript` instances to return `Task` instances instead of quadruples.

**Comment by riccardomurri** [comment]
cc @smaffiol and @arcimboldo : please review and state your opinion!

**Comment by arcimboldo** [comment]
I agree in having subcommands. Also, this is the current behavior of the SessionBasedDaemon class, for which we have a "server" and "client".

I am not sure merging gsession into session-based script is a good idea though, having a separate command makes more sense to me.
